coalib/results/ResultFilter.py

```diff
@@ -0,0 +1,220 @@
+
+    orig_result_diff_dict_dict = {}
+    for original_result in original_results:
+        orig_file_dict_copy = copy.deepcopy(original_file_dict)
+
+        for source_range in original_result.affected_code:
+            file_name = source_range.file
+            new_file = remove_range(orig_file_dict_copy[file_name],
+                                    source_range)
+            orig_file_dict_copy[file_name] = new_file
+
+        diff_dict = {}
+        for file_name in original_file_dict:
+            diff_dict[file_name] = Diff.from_string_arrays(
+                original_file_dict[file_name],
+                orig_file_dict_copy[file_name])
+
+        orig_result_diff_dict_dict[original_result] = diff_dict
+
+    # same thing for modified results. and yes this should really be in a
+    # function... but as long as both of these are not, code clone detection
+    # won't find them >:)
+    # fixme: yeah....
+
+    mod_result_diff_dict_dict = {}
+    for modified_result in modified_results:
+        mod_file_dict_copy = copy.deepcopy(modified_file_dict)
+
+        for source_range in modified_result.affected_code:
+            file_name = source_range.file
+            new_file = remove_range(mod_file_dict_copy[file_name],
+                                    source_range)
+            mod_file_dict_copy[file_name] = new_file
+
+        diff_dict = {}
+        for file_name in modified_file_dict:
+            diff_dict[file_name] = Diff.from_string_arrays(
+                modified_file_dict[file_name],
+                mod_file_dict_copy[file_name])
+
+        mod_result_diff_dict_dict[modified_result] = diff_dict
+
+    uniques = []
+
+    # THIS IS WHERE THE FUN BEGINS
+
+    for m_r in modified_results:
+        if True not in [basics_match(o_r, m_r) for o_r in original_results]:
+            uniques.append(m_r)
+
+        elif True not in [source_ranges_match(
+                original_file_dict,
+                modified_file_dict,
+                diffs_dict,
+                o_r,
+                orig_result_diff_dict_dict[o_r],
+                m_r,
+                mod_result_diff_dict_dict[m_r]) for o_r in original_results]:
+            uniques.append(m_r)
+
+    return uniques
+
+
+def basics_match(original_result,
+                 modified_result):
+    """
+    Checks whether the following properties of two results match:
+    * origin
+    * message
+    * severity
+    * debug_msg
+
+    :param original_result: A result of the old files
+    :param modified_result: A result of the new files
+    :return:                Boolean value whether or not the properties match
+    """
+    # origin might be a class or class name
+    original_origin = isinstance(original_result.origin, str) and \
+        original_result.origin or original_result.origin.__name__()
+    modified_origin = isinstance(modified_result.origin, str) and \
+        modified_result.origin or modified_result.origin.__name__()
```

Comment:
the origin is always a string, you can create this with a class for convenience but internally it's always a string